src/gsi/read_prepbufr.f90

```diff
@@ -1713,6 +1724,25 @@ subroutine read_prepbufr(nread,ndata,nodata,infile,obstype,lunout,twindin,sis,&
               end if
            end if
 
+!          If moisture ob (qob) and (2D/3D)RTMA, set tv flag information (based on tpc)
+!          regarding virtual vs. sensible temperaure, to get tdry (if virtual temp
+!          then compute tdry; if sensible temp, then tdry= tsen), then save tdry
+!          in q-obsdaig file for RTMA offline Auto-QC.
+           if (qob .and. (l_rtma3d .or. twodvar_regional)) then
+              tobs4q(1,:) = bmiss
+              tqm4q(1,:)  = bmiss
+              tvflg4q(1,:)= -one
+              do k=1,levs
+                 tvflg4q(1,k)=one                             ! initialize as sensible
+                 tobs4q(1,k)=obsdat(3,k)                      ! temp obs read in prepbufr
+                 tqm4q(1,k)=tqm(k)
+                 do j=1,20
+                    if (tpc(k,j)==vtcd) tvflg4q(1,k)=zero     ! reset flag if virtual
```

Comment:
General comment:  We should minimize the use of `==` when comparing floating point values.  This is done elsewhere in `read_prepbufr.f90` so no change needed.  Just a general comment.